README.md

```diff
@@ -1,2 +1,203 @@
+- rootbase module install procedure
+
+The routine requires the module specific manifest file and proceeds with it.
+
+root-get extensively makes use of cache to avoid re-building of entities. The already built packages are stored in cache directory.
+
+## External entity handling routine
+
+- External entity - Entity outside rootbase scope.
+
+The entity can be present in user system or may not be present on the system. For external entity also, root-get requires manifest file.
+The routine checks for entity presence in user system otherwise uses downloader routine to download the entity repository from GitHub(currently supported) and returns the path for the new downloaded directory. The further process takes the similar path as per internal entity routine.
+
+User can provide own manifest file with all neccessary attributes present in it to plug an external entity. To create your own manifest file please follow the instructions below
+
+- Creating module manifest file :
+```
+module:
+  name: 'modulename'
+  packageurl: 'link/to/github/repository'
+  tag: 0.0.0 
+  path: 'path/to/module/in/system'
+  publicheaders: 'if present'
+  sources: 'if present'
+  targets: 'modulename'
+  deps: 'dependencies'
+```
+
+- Example module.yml :
+```
+module: 
+  name: FFTW
+  packageurl: "https://github.com/root-project/root/tree/master/math/fftw"
+  tag: 0.0.0 
+  path: /home/username/root/math/fftw
+  publicheaders: 
+  sources: no_sources
+  targets: FFTW
+  deps: Core
+```  
+
+- Creating package manifest file :
+```
+package:
+  name: 
+  tag: 'version value'
+  targets:
+    target:
+    name: 'packagename'
+  products:
+    package:
+      name: 
+  module1:
+    name: 'modulename'
+    packageurl: 'link/to/github/repository'
+    tag: 0.0.0 
+    path: 'path/to/module/in/system'
+    publicheaders: 'if present'
+    sources: 'if present'
+    targets: 'modulename'
+    deps: 'dependencies'
+  module2:
+    .
+    .
+  module:
+```
+
+- Example packagename.yml :
+```
+package: 
+ name: "IO"
+ tag: 0.0.0
+ targets:
+  target:
+   name: "IO"
+ products:
+  package:
+   name: IO
+ module: 
+  name: "RCastor"
+  packageurl: "https://github.com/root-project/io/castor"
+  tag: 0.0.0
+  path: /home/ashwin/root/io/castor
+  publicheaders: 
+  sources: no_sources
+  targets: 
+  deps: Net;Core;RIO
+ module: 
+  name: "DCache"
+  packageurl: "https://github.com/root-project/io/dcache"
+  tag: 0.0.0
+  path: /home/ashwin/root/io/dcache
+  publicheaders: 
+  sources: no_sources
+  targets: 
+  deps: Net;Core;RIO 
+```
+
+## Prerequisites
+
+python3
+
+[matplotlib](https://matplotlib.org/users/installing.html)
+
+[networkx](https://networkx.github.io/documentation/stable/install.html)
+
+pyyaml (install with - sudo apt-get install python-yaml)
+
+cmake (Version 3.8 & above)
+
+
+## Initial setup
+Required - root build files directory, if not present already, do an out of source build for root
```

Comment:
Lets remove installation of ROOT base package from here (it is not really belongs to steps how to setup root-get)?

Reply:
I will move it in docs/setup-ROOTbase.md

Reply:
Sure, I'll create a separate file setup-ROOTbase.md and add there